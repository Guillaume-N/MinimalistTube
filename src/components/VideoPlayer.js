import React from "react";

const VideoPlayer = props => {
  const videoId = props.video;
  //  const videoId = props.match.params.id ? props.match.params.id : null;
  const url = `https://www.youtube.com/embed/${videoId}`;

  if (!props && !props.video && !props.video.etag) {
    return <div>select a video!</div>;
  }

  return (
    <section className="section">
      <div className="container">
        <div className="columns">
          <div className="column is-8 is-offset-2">
            <div className="video-container">
              <iframe title="video" src={url} />
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

export default VideoPlayer;
