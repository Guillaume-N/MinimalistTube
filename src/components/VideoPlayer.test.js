import React from "react";
import { shallow } from "enzyme";
import VideoPlayer from "./VideoPlayer";

describe("VideoPlayer Component", () => {
  it("should match snapshot", () => {
    const mockMatch = {
      params: {
        id: 1
      }
    };
    const wrapper = shallow(<VideoPlayer match={mockMatch} />);

    expect(wrapper).toMatchSnapshot();
  });
});
