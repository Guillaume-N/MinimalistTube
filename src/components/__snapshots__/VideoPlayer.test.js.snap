// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer Component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoPlayer
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="columns"
        >
          <div
            className="column is-8 is-offset-2"
          >
            <div
              className="video-container"
            >
              <iframe
                src="https://www.youtube.com/embed/1"
                title="video"
              />
            </div>
          </div>
        </div>
      </div>,
      "className": "section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="columns"
        >
          <div
            className="column is-8 is-offset-2"
          >
            <div
              className="video-container"
            >
              <iframe
                src="https://www.youtube.com/embed/1"
                title="video"
              />
            </div>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="column is-8 is-offset-2"
          >
            <div
              className="video-container"
            >
              <iframe
                src="https://www.youtube.com/embed/1"
                title="video"
              />
            </div>
          </div>,
          "className": "columns",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="video-container"
            >
              <iframe
                src="https://www.youtube.com/embed/1"
                title="video"
              />
            </div>,
            "className": "column is-8 is-offset-2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <iframe
                src="https://www.youtube.com/embed/1"
                title="video"
              />,
              "className": "video-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "https://www.youtube.com/embed/1",
                "title": "video",
              },
              "ref": null,
              "rendered": null,
              "type": "iframe",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="columns"
          >
            <div
              className="column is-8 is-offset-2"
            >
              <div
                className="video-container"
              >
                <iframe
                  src="https://www.youtube.com/embed/1"
                  title="video"
                />
              </div>
            </div>
          </div>
        </div>,
        "className": "section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="columns"
          >
            <div
              className="column is-8 is-offset-2"
            >
              <div
                className="video-container"
              >
                <iframe
                  src="https://www.youtube.com/embed/1"
                  title="video"
                />
              </div>
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="column is-8 is-offset-2"
            >
              <div
                className="video-container"
              >
                <iframe
                  src="https://www.youtube.com/embed/1"
                  title="video"
                />
              </div>
            </div>,
            "className": "columns",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="video-container"
              >
                <iframe
                  src="https://www.youtube.com/embed/1"
                  title="video"
                />
              </div>,
              "className": "column is-8 is-offset-2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <iframe
                  src="https://www.youtube.com/embed/1"
                  title="video"
                />,
                "className": "video-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": "https://www.youtube.com/embed/1",
                  "title": "video",
                },
                "ref": null,
                "rendered": null,
                "type": "iframe",
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
